### PR TITLE
Bump RKE2 to v1.23.11-rc1+rke2r1

### PR DESCRIPTION
**Pull request by bk201**
Related issue: https://github.com/harvester/harvester/issues/2785

### CODE REVIEW

**Review by Vicente-Cheng — APPROVED** [review]
test with some basic operations like creating volume/VM/image, attach/detach, and VM migration. Looks good.
thanks!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

File: scripts/version-rke2 [hunk 1]
```diff
@@ -1 +1 @@
-RKE2_VERSION="v1.22.12+rke2r1"
+RKE2_VERSION="v1.23.11-rc1+rke2r1"
```
Comment (guangbochen) [inline scripts/version-rke2:1]: Q: do we plan to bump to a stable `rke2+v1.23.11` version before the v1.1.0 release?

Comment (bk201) [reply scripts/version-rke2:1]: Need to be v1.23.12 due to a regression.
I'd like to first stick to v1.23.10 first. I'll update the PR.

**Comment by bk201** [comment]
Replace the PR with https://github.com/harvester/harvester-installer/pull/342 (different source branch name).

**Pull request closed without merging**